infrastructure/smart-contracts/TestCases.md

```diff
@@ -0,0 +1,138 @@
+            kycAdmin,
+            xaiDefaultAdmin,
+            xaiMinter,
+            esXaiDefaultAdmin,
+            esXaiMinter,
+            gasSubsidyDefaultAdmin,
+            gasSubsidyTransferAdmin,
+            nodeLicenseDefaultAdmin,
+            addr1,
+            addr2,
+            addr3,
+            addr4,
+            operator,
+            rollupController,
+            tiers,
+            secretKeyHex,
+            publicKeyHex: "0x" + publicKeyHex,
+            referee: referee9,
+            nodeLicense: nodeLicense8,
+            poolFactory: poolFactory2,
+            gasSubsidy,
+            esXai: esXai2,
+            xai,
+            rollupContract
+        }
+```
+
+Below that, you'll see the individual tests that reference all of the different test files. Each test receives the "deployInfrastructure" as a param. The deployInfrastructure object makes all of the objects/contracts/signers available in the test files.
+
+``` javascript
+  describe("Xai", XaiTests(deployInfrastructure).bind(this));
+  describe("EsXai", esXaiTests(deployInfrastructure).bind(this));
+  describe("Node License", NodeLicenseTests(deployInfrastructure).bind(this));
+  describe("Referee", RefereeTests(deployInfrastructure).bind(this));
+```
+
+## Running the tests
+
+In order to run the tests you must ensure everything has been installed and built properly.
+
+1. Confirm that you have a .env with a valid MNEMONIC.
+2. From the root directory run `pnpm install`
+3. Tests will use core, so core has to be built locally with `pnpm -filter @sentry/core run build`.
+5. Update the `hardhat.config.cjs` file. Change the default network on line 14 to use `hardhat` instead of the current `arbitrumOne`.
+6. Inside of `infrastructure/smart-contacts`, run `pnpm local` to start a local hardhat node.
+7. Open a second terminal/command prompt and CD into `infrastructure/smart-contacts`
+8. Run `pnpm test` to run the test suite.
+
+## Adding New Tests
+
+Tests for new functionality should have new files/folders created and then those test functions should be called from the main Fixture.mjs file.
+
+## Adding Contracts
+
+When a contract is added , you will need to update the Fixture.mjs file to include the deployment of your new contract. You should also add it to the return object at the end of the file so that it may be consumed in other test files.
+
+## Upgrading Contracts
+
+When a contract is upgraded, you will need to update the Fixture.mjs file to include the upgrade of your new contract. You should also update the return object at the end of the file for the new upgraded contract so that it may be consumed in the other test files.
+
+## Test Considerations
+
+Tests that require challenge submissions/assertions/claims have a few special considerations that need to be accounted for.
+
+1. You must submit an initial challenge before creating a stake pool, or the stake pool creation will fail due to subtraction overflow with the challenge number being 0.
```

Comment:
We might want to create a story and resolve this now by creating the initial challenge in the fixture ?

Reply:
I'm neutral on this, adding to the fixture would require updating all existing tests that use challenges as the challenge numbers would all change. I don't think it's difficult to submit a test challenge with the helper function and it gives more fine grained control over what confirm data is used.